File: src/main/java/com/bank/service/internal/TimePolicy.java
package com.bank.service.internal;

import com.bank.domain.Time;

public class TimePolicy {
    private Time time;

    public TimePolicy(Time time) {
        this.time = time;
    }

    public boolean isTimeValid() {
        boolean isAfter = time.getCurrentDate().isAfter(time.getStartDate());
        boolean isBefore = time.getCurrentDate().isBefore(time.getEndDate());
        return isAfter && isBefore;
    }
}

File: src/main/java/com/bank/service/internal/TimePolicy.kt
package com.bank.service.internal

import com.bank.domain.Time

open class TimePolicy constructor(private val time: Time) {

    fun isTimeValid(): Boolean {
        val isAfter = time.currentTime.isAfter(time.startTime)
        val isBefore = time.currentTime.isBefore(time.endTime)
        return isAfter && isBefore
    }
}